YubiKit/YubiKit/Connection/SmartCard/NFCSmartCardConnection.swift

```diff
@@ -345,6 +345,39 @@ private final class NFCConnectionManager: NSObject, @unchecked Sendable {
     func stop(with result: Result<String?, Error>, completion: @escaping @Sendable () -> Void) {
         /* Fix trace: trace(message: "Manager.stop(with:) - result: \(String(describing: result))") */
 
+        // If already inactive, complete immediately
+        guard currentState.phase != .inactive else {
+            completion()
+            return
+        }
+
+        // If already stopping, just chain the completion
+        if currentState.phase == .stopping {
+            if let existing = currentState.stopCompletion {
+                currentState.stopCompletion = {
+                    existing()
+                    completion()
+                }
+            } else {
+                currentState.stopCompletion = completion
+            }
+            return
+        }
```

Comment:
Isn't this just duplicating the block below it? Would it make sense to replace this with an "if .stopping: return" below instead?